pypsa/plot/maps/interactive.py

```diff
@@ -323,279 +320,875 @@ def iplot(
+            missing_columns = [col for col in extra_columns if col not in df.columns]
+            valid_columns = [col for col in extra_columns if col in df.columns]
+
+            if missing_columns:
+                msg = (
+                    f"Columns {missing_columns} not found in {component}. "
+                    f"Using only valid columns: {valid_columns}."
+                )
+                logger.warning(msg)
+
+            layer_columns.extend(valid_columns)
+            self._tooltip_columns = list(set(self._tooltip_columns + valid_columns))
+
+        df = df[layer_columns].copy()
+        df.index.name = "name"
+
+        if extra_columns:
+            # Round all numeric columns to 3 decimal places. Only columns in valid_columns
+            numeric_cols = [
+                col for col in valid_columns if pd.api.types.is_numeric_dtype(df[col])
+            ]
+            for col in numeric_cols:
+                df[col] = df[col].round(3)
+
+        return df
+
+    # Layer functions
+    def add_bus_layer(
+        self,
+        bus_sizes: float | dict | pd.Series = 25000000,
+        bus_colors: str | dict | pd.Series = "cadetblue",
+        bus_alpha: float | dict | pd.Series = 0.7,
+        bus_columns: list | None = None,
+    ) -> None:
+        """Add a bus layer of Pydeck type ScatterplotLayer to the interactive map.
+
+        Parameters
+        ----------
+        bus_sizes : float/dict/pandas.Series
+            Sizes of bus points in radius² (meters²), defaults to 25000000.
+        bus_colors : str/dict/pandas.Series
+            Colors for the buses, defaults to 'cadetblue'.
+        bus_alpha : float/dict/pandas.Series
+            Add alpha channel to buses, defaults to 0.7.
+        bus_columns : list, default None
+            List of bus columns to include. If None, only the bus index and x, y coordinates are used.
+            Specify additional columns to include in the tooltip.
+
+        """
+        # Check if columns exist and only keep the ones that also exist in the network
+        bus_data = self.prepare_component_data(
+            "Bus",
+            default_columns=["x", "y"],
+            extra_columns=bus_columns,
         )
 
-        gdf_links[gdf_links.is_valid].explore(
-            m=map, color=link_colors, tooltip=tooltip, popup=popup, name="Links"
+        # For default tooltip
+        bus_data["value"] = _convert_to_series(bus_sizes, bus_data.index).round(3)
+        bus_data["coords"] = bus_data[["x", "y"]].apply(
+            lambda row: f"({row['x']:.3f}, {row['y']:.3f})", axis=1
         )
-        components_present.append("Link")
 
-    if not n.c.buses.static.empty and "Bus" in components:
-        gdf_buses = gpd.GeoDataFrame(
-            n.c.buses.static,
-            geometry=gpd.points_from_xy(n.c.buses.static.x, n.c.buses.static.y),
-            crs=crs,
+        # Map bus sizes
+        bus_data["radius"] = bus_data["value"] ** 0.5
```

Comment:
It could be dangerous if using the rounded values from above (for some edge cases with very low values).